log/stat/stat.go

```diff
@@ -0,0 +1,112 @@
+package stat
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+	"text/tabwriter"
+
+	"github.com/peak/s5cmd/strutil"
+)
+
+const (
+	totalCount = iota
+	succCount
+)
+
+var (
+	enabled bool
+	stats   statistics
+)
+
+type statistics [2]syncMap
+
+// InitStat initializes collecting program statistics.
+func InitStat() {
+	enabled = true
+	for i := range stats {
+		stats[i] = syncMap{
+			Mutex:       sync.Mutex{},
+			mapStrInt64: map[string]int64{},
+		}
+	}
+}
+
+// syncMap is a statically typed and synchronized map.
+type syncMap struct {
+	sync.Mutex
+	mapStrInt64 map[string]int64
+}
+
+func (s *syncMap) add(key string, val int64) {
+	s.Lock()
+	defer s.Unlock()
+
+	s.mapStrInt64[key] += val
+}
+
+// Stat is for storing a particular statistics.
+type Stat struct {
+	Visited    string `json:"operation"`
```

Comment:
Is it `operation` or `Visited`?  I'm confused.

Reply:
since there is no restriction for from where `stat collect` might be called, `visited` is probably ok option, but currently `stat collect` is being called only from within operations; `operation` seems correct one. thanks for pointing it out.